docs/multinodejanusgraphcluster.adoc

```diff
@@ -0,0 +1,110 @@
+
+2. Configure each node to use a `JanusGraphChannelizer`, which injects lower-level Gremlin Server components, like the GremlinExecutor, into the JanusGraph project, giving us greater control of the Gremlin Server.
+
+To configure each node to use a `JanusGraphChannelizer`, we must update the `gremlin-server.yaml` to do so:
+
+[source, properties]
+----
+channelizer: org.janusgraph.channelizers.JanusGraphWebSocketChannelizer
+----
+
+There are a few channelizers you can choose from:
+
+1. org.janusgraph.channelizers.JanusGraphWebSocketChannelizer
+2. org.janusgraph.channelizers.JanusGraphHttpChannelizer
+3. org.janusgraph.channelizers.JanusGraphNioChannelizer
+4. org.janusgraph.channelizers.JanusGraphWsAndHttpChannelizer
+
+All of the channelizers share the exact same functionality as their TinkerPop counterparts.
+
+[[tinkerpop-with-remote]]
+===== Using TinkerPop's withRemote Functionality
+
+Since traversal references are bound on the JanusGraph servers, we can make use of http://tinkerpop.apache.org/docs/current/reference/#connecting-via-remotegraph[TinkerPop's withRemote functionality]. This will allow one to run gremlin queries locally, against a remote graph reference. Traditionally, one runs queries against remote Gremlin Servers by sending String script representations, which are processed on the remote server and the response serialized and sent back. However, TinkerPop also allows for the use of `remoteGraph`, which could be useful if you are building a TinkerPop compliant graph infrastructure that is easily transferable to multiple implementations.
+
+
+To use this functionality in JanusGraph, we must first ensure we have created a graph on the remote JanusGraph cluster:
+
+```
+ConfiguredGraphFactory.create("graph1");
+```
+
+Next, we must wait 20 seconds to ensure the traversal reference is bound on every JanusGraph node in the remote cluster.
+
+Finally, we can locally make use of the `withRemote` method to access a local reference to a remote graph:
+
+[source, gremlin]
+----
+gremlin> cluster = Cluster.open('conf/remote-objects.yaml')
+==>localhost/127.0.0.1:8182
+gremlin> graph = EmptyGraph.instance()
+==>emptygraph[empty]
+gremlin> g = graph.traversal().withRemote(DriverRemoteConnection.using(cluster, "graph1_traversal"))
+==>graphtraversalsource[emptygraph[empty], standard]
+----
+
+For completion, the above `conf/remote-objects.yaml` should tell the `Cluster` API how to access the remote JanusGraph servers; for example, it may look like:
+
+[source, properties]
+----
+hosts: [remoteaddress1.com, remoteaddress2.com]
+port: 8182
+username: admin
+password: password
+connectionPool: { enableSsl: true }
+serializer: { className: org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0,
+              config: { ioRegistries: [org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerIoRegistryV3d0] }}
```

Comment:
This should match what's in the JanusGraph version of [`remote-objects.yaml`](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-dist/src/assembly/static/conf/remote-objects.yaml#L3):

```
serializer: { className: org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV1d0,
             config: { ioRegistries: [org.janusgraph.graphdb.tinkerpop.JanusGraphIoRegistry] }}
```